tests/fixtures/core/current_user.py

```diff
@@ -0,0 +1,8 @@
+import pytest
+
+from core import current_user
+
+
+@pytest.fixture
+def _set_current_user(user):
```

Comment:
```suggestion
def current_user(mocker, user):
  return mocker.patch("core.current_user, get_current_user", return_value=user)
```

Эта фикстура будет гораздо лучше:

1. Она исключает из тестового набора логику работы с тредами, которую нет смысла использовать, если мы её не тестируем
2. Она более универсальна с точки зрения API, т.к. её всегда можно призвать и самому поменять значение current_user

Reply:
Надо патчить модуль, откуда функция вызывается, а не где она определена. Вот так работает:

```python
import pytest


@pytest.fixture
def current_user(mocker, user):
    return mocker.patch("apps.orders.services.order_refunder.get_current_user", return_value=user)
```

Но это не похоже на кор-фикстуру и для других сервисов нужно писать другие фикстуры.

Reply:
> Надо патчить модуль, откуда функция вызывается, а не где она определена.

Ты проверяла это на нашем коде?

Reply:
Да, то, что я скинула, рабочий код в нашем коде. Твой не работает ;(

Reply:
Если поставить точку вместо запятой здесь `current_user, get_current_user` — то тоже работать не будет.

Reply:
@f213, тогда делаю две отдельные фикстуры: одна для мока в рефанд-сервисе, вторая — для пэйд-сеттер.

Reply:
Сойдёт. А не хочешь сделать фикстуру _set_current_user() с autouse=True, и чтобы она устаналвивала текущего юзера только если включена марка auditlog? Вряд ли нам понадобится auditlog без текущего юзера.

Reply:
> А не хочешь сделать фикстуру _set_current_user() с autouse=True, и чтобы она устаналвивала текущего юзера только если включена марка auditlog?

Хочу, только из-за того что эта фикстура используется [не только](https://github.com/tough-dev-school/education-backend/blob/db5f5bf157064feff8b34f378e8942624f822dbb/tests/apps/orders/services/tests_order_refunder.py#L34) в аудитлог-режиме, я не стала в ней самой делать ветвлений; я сделала иначе, мне кажется, так универсальнее:

<img width="757" alt="image" src="https://github.com/tough-dev-school/education-backend/assets/67327570/80da98db-c56e-4270-bffe-93960329d214">

Это оставляет возможность всё-таки при необходимости использовать `_set_current_user` отдельно от аудитлога.

(на скрине фикстурный модуль `pause_slow_background_jobs.py`, который я переименовала в `markers.py`, чтобы все настройки маркеров были в одном месте, а не стоп тасок в одном, мок функций в другом, подключение фикстур в третьем)

Reply:
> Вряд ли нам понадобится auditlog без текущего юзера.

Надобится ;)

<img width="707" alt="image" src="https://github.com/tough-dev-school/education-backend/assets/67327570/fb171c9e-372b-4ba8-8035-5f8305c866e4">

Но это крайний случай.